### PR TITLE
feat: support exp and refactor pow

### PR DESCRIPTION
**Pull request by lmatz**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

As per the title. Since only float64 is supported, let the cast happen in frontend.

## Checklist

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [x] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features). #7934
- [x] I have demonstrated that backward compatibility is not broken by breaking changes and created issues to track deprecated features to be removed in the future. (Please refer to the issue)
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Documentation

~- [ ] My PR **DOES NOT** contain user-facing changes.~

<details><summary>Click here for Documentation</summary>

### Types of user-facing changes

Please keep the types that apply to your changes, and remove the others.

- SQL commands, functions, and operators

### Release note

Support function `exp`. The decimal version is not supported.

</details>
#7725

### CODE REVIEW

File: src/tests/regress/data/expected/float8.out [hunk 1]
```diff
@@ -616,7 +616,15 @@ ERROR:  cannot take logarithm of zero
 SELECT ln(f.f1) from FLOAT8_TBL f where f.f1 < '0.0' ;
 ERROR:  cannot take logarithm of a negative number
 SELECT exp(f.f1) from FLOAT8_TBL f;
-ERROR:  value out of range: underflow
+          exp          
+-----------------------
+                     1
+                     0
+                     0
+ 7.399123060905129e-16
```
Comment (lmatz) [inline src/tests/regress/data/expected/float8.out:624]: both float64, but RW works while PG underflows....?

Comment (xiangjinwu) [reply src/tests/regress/data/expected/float8.out:624]: I'd prefer comment this test case out (`--@ ` in input `.sql` file) rather than updating its expected output, which may give a wrong signal of being compatible.

As for this issue, PG does have some overflow/underflow checking that we lack: #6826

BTW the `select exp(-10000000);` test case (and float variant `select exp(-10000000::float8);`) in sqllogictest also underflows in PG. But I think we can keep it there and acknowledge in comments it is not PG-compatible right now.

To summarize, `sqllgocitest` to document our current behavior and `regress` tests to keep record of our compatibility. Does this make sense?

File: src/frontend/src/binder/expr/function.rs [hunk 2]
```diff
@@ -331,6 +331,7 @@ impl Binder {
                 ("ceil", raw_call(ExprType::Ceil)),
                 ("floor", raw_call(ExprType::Floor)),
                 ("abs", raw_call(ExprType::Abs)),
+                ("exp", raw_call(ExprType::Exp)),
```
Comment (xiangjinwu) [inline src/frontend/src/binder/expr/function.rs:334]: PG allows spelling both `pow` and `power`. We can add one line here to dispatch `power` to `ExprType::Pow`.

File: src/expr/src/vector_op/exp.rs [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+use num_traits::Float;
+use risingwave_common::types::OrderedF64;
+
+use crate::{ExprError, Result};
+
+pub fn exp_f64(input: OrderedF64) -> Result<OrderedF64> {
+    let res = input.exp();
+    if res.is_infinite() {
+        Err(ExprError::NumericOutOfRange)
```
Comment (xiangjinwu) [inline src/expr/src/vector_op/exp.rs:23]: FYI PG logic [here](https://github.com/postgres/postgres/blob/REL_15_2/src/backend/utils/adt/float.c#L1649) if we want to be fully compatible.

**Review by xiangjinwu — APPROVED** [review]
rest lgtm

File: src/tests/regress/data/expected/float8.out (hunk 1 above)
Comment (lmatz) [reply src/tests/regress/data/expected/float8.out:624]: I see, it makes sense, thanks for the detailed explanation

File: src/expr/src/vector_op/exp.rs (hunk 3 above)
Comment (lmatz) [reply src/expr/src/vector_op/exp.rs:23]: let me track it in a new issue, I generally feel that If RW is more `powerful` than PG, it is ok 🤔 
Will correct it when users complain
